chrome/content/zotero/itemTree.jsx

```diff
@@ -693,26 +693,29 @@ var ItemTree = class ItemTree extends LibraryTree {
 							if (collectionTreeRow.isUnfiled() && item.getCollections().length) {
 								this._removeRow(row);
 							}
+							// If moved from under another item to top level, remove old row and add new one
+							else if (parentIndex != -1 && !parentItemID) {
+								this._closeContainer(row);
+								this._removeRow(row);
+	
+								let beforeRow = this.rowCount;
+								this._addRow(new ItemTreeRow(item, 0, false), beforeRow);
+	
+								sort = true;
+							}
```

Comment:
Oh, but the problem here is that I changed the meaning of the "Top-level item" block in a7964f7116fc0a386710f51ad55805d4a5c54246. It's supposed to be for top-level items being changed, and it was that way because the `row` in `this.isContainer(row)` was coming from the row map that hadn't been updated yet.

So the proper fix here is a reverting to `this.isContainer(row)` and adding an additional condition to prevent that from applying to attachments, which have become containers.

Reply:
Haven't tested, but just `this.getLevel(row) === 0`, maybe?

Reply:
`parentIndex == -1` would also work, because that's just testing for level being 0.

Reply:
Oh, and we're using `parentIndex == -1` below. So we'd need to combine with the following block and add a sub-block for the `parentItemID` test.

Reply:
(And then we can remove the `parentIndex != -1` from the following conditions, since it will be redundant.)

Reply:
I reverted to `this.isContainer(row)` but then I realized that now handling of moving a row between/to/from top level items should apply equally to containers (attachments) and to non-containers (notes). So I refactored this to handle change in parent regardless of if it is an container or not, with the Unfiled items being a special case at the bottom. Does this make sense?

I added test cases for moving between/to/from top-level items of attachments and notes. I checked and with earlier iterations (before a7964f7116fc0a386710f51ad55805d4a5c54246, after, and with the initial commit from this PR), at least one of them would be failing.